packages/node_modules/@ciscospark/internal-plugin-wdm/src/device/device.js

```diff
@@ -282,6 +351,27 @@ const Device = SparkPlugin.extend({
     }
   },
 
+  _updateServiceCatalog(newRegistration) {
+    Object.keys(newRegistration.services).forEach((service) => {
+      const uri = newRegistration.services[service];
+      const u = Url.parse(uri);
+      const hosts = newRegistration.serviceHostMap.hostCatalog[u.host];
+      const existingService = this.serviceCatalog.get(service);
+      if (existingService) {
```

Comment:
Oh, thanks! I overlook the fact that you can add remove, add, merge as an option to set. Looks like it does what I want by adding the option `{remove: false}` since the I wanted to add/merge here.